### PR TITLE
[Perf] Cache rotary position IDs in Qwen2-VL vision encoder

### PR DESCRIPTION
**Pull request by Inference1**
## Purpose

`Qwen2VisionTransformer.rot_pos_emb` recomputes the spatial position grids (`hpos_ids` / `wpos_ids`) on every forward call. For video inputs where consecutive frames share the same resolution, this is redundant work on the CPU hot path.

Extract the grid computation into `rot_pos_ids(h, w, spatial_merge_size)` as a `@staticmethod` with `@lru_cache(maxsize=1024)`. Switch the implementation from torch ops to numpy `broadcast_to` + `reshape` + `transpose` so the args are plain ints (hashable without wrappers) and no GPU tensor is allocated for what is fundamentally a CPU index computation. The caller becomes a one-liner with `torch.cat` + `.to(device, non_blocking=True)`.

## Test Plan
test_rot_pos_cache.py — correctness + micro-benchmark
test_e2e_perf.py — E2E with Qwen2.5-VL-3B
[test_rot_pos_cache.py](https://github.com/user-attachments/files/27514189/test_rot_pos_cache.py)
[test_e2e_perf.py](https://github.com/user-attachments/files/27514191/test_e2e_perf.py)

```bash
python tests/test_rot_pos_cache.py
```

Correctness: compares cached numpy path against original torch path for 6 spatial configs typical of Qwen2.5-VL:
```
(h,w,sms): (14,14,2), (28,28,2), (14,28,2), (56,56,2), (28,14,2), (42,28,2)
```

Performance: 500 iters x 3 configs `[(14,14,2), (28,28,2), (56,56,2)]`, measures wall time for baseline vs cache-hit.

E2E with Qwen2.5-VL-3B-Instruct (any variant works), single image 448x448, 4 GPU TP:
```bash
CUDA_VISIBLE_DEVICES=4,5,6,7 python tests/test_e2e_perf.py
```

## Test Result

```
=== Correctness ===
  PASS (6 configs, exact match)

=== Cache identity ===
  PASS (same args -> same object)

=== Performance (500 iters x 3 configs) ===
  baseline (torch recompute):   178.2 us/iter
  patched  (lru_cache hit):       0.3 us/iter
  speedup:                      ~700x

=== E2E (Qwen2.5-VL-3B-Instruct, 448x448, 4xH200) ===
  rot_pos_ids micro:    51 us -> 0.08 us  (666x)
  rot_pos_emb full:    167 us -> 77 us    (2.2x, dominated by cos/sin gather)
  E2E single-image:    ~42 ms both (GPU-bound, rot_pos_ids fraction negligible)
```

Single-image latency is GPU-bound so the saving doesn't surface in E2E numbers. The win matters for video workloads where `rot_pos_emb` is called N_frames times with identical (h, w) -- cumulative CPU overhead drops from milliseconds to microseconds.

### CODE REVIEW

**Comment by Inference1** [comment]
Attached reproduction scripts for the test results above:       
  - `test_rot_pos_cache.py` — correctness + micro-benchmark                          
  - `test_rot_pos_e2e_perf.py` — E2E with Qwen2.5-VL-3B   
[test_rot_pos_cache.py](https://github.com/user-attachments/files/27514189/test_rot_pos_cache.py)
[test_rot_pos_cache.py](https://github.com/user-attachments/files/27514189/test_rot_pos_cache.py)
[test_e2e_perf.py](https://github.com/user-attachments/files/27514191/test_e2e_perf.py)
[test_e2e_perf.py](https://github.com/user-attachments/files/27514191/test_e2e_perf.py)

@sighingnow @vadiklyutiy Could you add the ready label when you have a chance to review? Thanks!

**Comment by ywang96** [comment]
We don't want to encourage bot-generated PRs and this particular one is adding codes to an outdated model architecture. 

If you indeed would like this feature, please feel free to fork vLLM and patch it yourself. Thanks!
